### ISSUE TITLE
How create data.t7 from imageset

### ISSUE THREAD

**Issue by ilichev-andrey** [torch/torch7#854]
Hello.

I have images in two folders:
each image - 96x96x3
- animals
- peoples

How create model in format .t7 from two classes?

I do for the neural network training differences between man and animal.
After creating a model, I plan to train on Deep Learning with Torch
Any reply will be important for me.

**Comment by SergeyMalashenko** [comment]
t7 is no data format. This is just the file extension
The file is created when you call torch.save (...).
Then the file can be loaded with torch.load(...)

In my project all datasets have the following structure
{
data  : torch.Tensor(1000,3,96,96)  
label : torch.Tensor(1000,2)              ( 1 - animal, 2 - people)
} : Lua table

**Comment by lvdong183** [comment]
I meet with the same difficulties. 
If you have an answer. Could you share your method with me?Thanks a lot!

**Comment by SergeyMalashenko** [comment]
All depends from you. Only you decide how to store your data. 
In my project I use the following format
dataset = { data = sample_s, label = label_s )
and the I call
torch.save( "output.t7", dataset )

If you use an external dataset you can load it with
dataset = torch.load( "output.t7")
print( dataset )

**Comment by Jayhello** [comment]
load:      Readed_t7 = torch.load('cifar10-train.t7')
print(Readed_t7)
{
  data : ByteTensor - size: 10000x3x32x32
  label : ByteTensor - size: 10000
}
print(Readed_t7.data[1][1])    => 输出一个 32*32的矩阵
print(Readed_t7.label[1])      =>  for example : 4
data = torch.Tensor(3,3,3):fill(1)
(1,.,.) = 
  1  1  1
  1  1  1
  1  1  1

(2,.,.) = 
  1  1  1
  1  1  1
  1  1  1

(3,.,.) = 
  1  1  1
  1  1  1
  1  1  1
[torch.DoubleTensor of size 3x3x3]

label=torch.Tensor(1):fill(1)

data_to_write = {data=data, label=label}
torch.save("/tmp/saved_data.t7", data_to_write)
 testt7 = torch.load('/tmp/saved_data.t7.t7')
 print(testt7)
{
  data : DoubleTensor - size: 3x3x3
  label : DoubleTensor - size: 1
}

**Comment by crazyzsy** [comment]
@SergeyMalashenko ，Can you tell me  what sample_s is?

**Comment by SergeyMalashenko** [comment]
@crazyzsy  in my case sample_s is a set of images.

**Comment by pesciolina** [comment]
remember how you created the file.t7 to load on torch for the image folders,please?

**Comment by abduallahnoorsayket** [comment]
Excuse me !!   How can I convert a .t7 file into  image file in python ?? Anyone    please help me ?? i am in confusion

**Comment by SergeyMalashenko** [comment]
You need to use pyTorch for this
https://stackoverflow.com/questions/41861354/loading-torch7-trained-models-t7-in-pytorch
